flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDeserializationSchema.java

```diff
@@ -242,11 +247,33 @@ private DeserializationRuntimeConverter wrapIntoNullableConverter(Deserializatio
 			return Optional.of(createObjectArrayConverter(((BasicArrayTypeInfo) typeInfo).getComponentInfo()));
 		} else if (isPrimitiveByteArray(typeInfo)) {
 			return Optional.of(createByteArrayConverter());
+		} else if (typeInfo instanceof MapTypeInfo) {
+			MapTypeInfo<?, ?> mapTypeInfo = (MapTypeInfo<?, ?>) typeInfo;
+			return Optional.of(createMapConverter(mapTypeInfo.getKeyTypeInfo(), mapTypeInfo.getValueTypeInfo()));
 		} else {
 			return Optional.empty();
 		}
 	}
 
+	private DeserializationRuntimeConverter createMapConverter(TypeInformation keyType, TypeInformation valueType) {
+		DeserializationRuntimeConverter valueConverter = createConverter(valueType);
+		DeserializationRuntimeConverter keyConverter = createConverter(keyType);
+
+		return (mapper, jsonNode) -> {
+			// ObjectNode stores Map<String, JsonNode> internally, so we can only get String keys.
+			Iterator<String> fieldNames = jsonNode.fieldNames();
+			Map<Object, Object> result = new HashMap<>();
+			while (fieldNames.hasNext()) {
+				String stringKey = fieldNames.next();
+				JsonNode keyNode = TextNode.valueOf(stringKey);
```

Comment:
We can use the lambda:
```java
fieldNames.stream().map() ...
```
to construct the result Map.

Reply:
done.